src/dune_pkg/opam_repo.ml

```diff
@@ -58,25 +58,36 @@ module Source = struct
              we need to look up *)
           Rev_store.mem rev_store ~rev:ref
           >>= (function
-           | true -> Fiber.return @@ Some (Commitish.Commit ref)
+           | true ->
+             (* CR-Leonidas-from-XIV is this always a commit? *)
+             Fiber.return @@ Some (Commitish.Commit ref)
            | false ->
              Rev_store.ref_type rev_store ~source:url ~ref
-             >>| (function
-              | Some `Tag -> Some (Commitish.Tag ref)
-              | Some `Head -> Some (Commitish.Branch ref)
+             >>= (function
+              | Some `Tag -> Fiber.return @@ Some (Commitish.Tag ref)
+              | Some `Head -> Fiber.return @@ Some (Commitish.Branch ref)
               | None ->
-                User_error.raise
-                  ~loc
-                  ~hints:
-                    [ Pp.text
-                        "Make sure the URL is correct and the repository contains the \
-                         branch/tag"
-                    ]
-                  [ Pp.textf
-                      "Opam repository at '%s' does not have a reference '%s'"
-                      url
-                      ref
-                  ]))
+                (* we have to update the local repo and see if the commit exists *)
+                let* _ =
```

Comment:
It's important to annotate the types of anything that we're ignoring. Although I'm not sure how useful is `Remote.update` anymore.

Reply:
Good point.

I would argue it is still useful to have a way to `don't_update`. It could be merged into `add_repo`, but it feels like mixing responsibilities. Maybe a refactor that would change `add_repo` to imply working with a remote source instead of implying it will add a repository.